### ISSUE TITLE
Published messages received by a realtime client have an empty connection ID

### ISSUE THREAD

**Issue by Rosalita** [ably/ably-go#484]
When messages are received by a realtime client, the message connection ID is empty.

I wrote a small script to recreate this issue which creates a realtime client, subscribes to a channel, then publishes the following to that channel:
- a message from a realtime client
- a message from a REST client
- a message from a REST client on behalf of a realtime client (using the realtime client's connection key)

```
package main

import (
	"context"
	"fmt"
	"os"
	ably "github.com/ably/ably-go/ably"
	"time"
)

func main() {
	key, _ := os.LookupEnv("ABLY_PRIVATE_KEY")
	ctx := context.Background()

	// Realtime client
	realtimeClient, err := ably.NewRealtime(ably.WithKey(key))
	if err != nil {
		fmt.Println(err)
	}

	realtimeChannel := realtimeClient.Channels.Get("test")
	realtimeChannel.SubscribeAll(ctx, handleMessage)
	realtimeChannel.Publish(ctx, "realtime publish", "hello")

	//REST client
	restClient, _ := ably.NewREST(
		ably.WithKey(key),
	)

	restChannel := restClient.Channels.Get("test")
	restChannel.Publish(ctx, "rest publish", "hello")
	restChannel.Publish(ctx, "rest publish on behalf of realtime", "hello", ably.PublishWithConnectionKey(realtimeClient.Connection.Key()))
	
	// wait for message handling
	time.Sleep(1 * time.Second)
}

func handleMessage(msg *ably.Message) {
	fmt.Printf("Received message: name=%q data=%q connectionID=%q\n", msg.Name, msg.Data, msg.ConnectionID)
}
```

output after running this script is 

```
Received message: name="realtime publish" data="hello" connectionID=""
Received message: name="rest publish" data="hello" connectionID=""
Received message: name="rest publish on behalf of realtime" data="hello" connectionID=""
```

Expected behaviour:
Expected that connectionID would be populated with the connectionID of the publishing client.

Client library specification (RTL6f):
Message#connectionId should match the current Connection#id for all published messages, a test should exist to ensure the connectionId for received messages matches that of the publisher.

https://docs.ably.io/client-lib-development-guide/features/#RTL6f

┆Issue is synchronized with this [Jira Bug](https://ably.atlassian.net/browse/SDK-1705) by [Unito](https://www.unito.io)
┆sprint: SDK 2022-Q1-S8

**Comment by Rosalita** [comment]
I ran this script against branch `fix#58` and it looks like this PR https://github.com/ably/ably-go/pull/481 has partially fixed this issue. The output from the script on that branch was: 

```
Received message: name="realtime publish" data="hello" connectionID="abjXusAo7F"
Received message: name="rest publish" data="hello" connectionID=""
Received message: name="rest publish on behalf of realtime" data="hello" connectionID="abjXusAo7F"
```

It looks like there's still some more work to do to populate the message connectionID when the message is published by a REST client.

**Comment by Morganamilo** [comment]
Looking at the spec and js library there appears to be no mention of a connection ID in rest. So I'm assuming it's intended for that one to have no client ID. Other two cases are fixed so this is not a bug.
